### ISSUE TITLE
[FEATURE]: [Mollie] Currency Unit Conversion

### ISSUE THREAD

**Issue by Sakilmostak** [juspay/hyperswitch#2231]
### :memo: Feature Description

- Each currency can be described in terms of base or minor units. 
- For instance, when dealing with USD currency, the Cent is considered as the minor unit, while the Dollar is considered the base unit. 
- In Hyperswitch, the amount value is expected to be always provided in minor units. 
- For example: In USD, If the amount is 1 in base unit (Dollar), then it will be equivalent to 100 in minor units (Cent)
- Some of the connectors integrated require the amount to be converted to a desirable unit before being passed to them. 
- We have functions `to_currency_base_unit` and `to_currency_lower_unit` in place to convert the minor unit amount to its decimal equivalent and vice versa. These conversions are handled explicitly based on the connector.

### :hammer: Possible Implementation

- ConnectorCommon trait have been implemented for the connector. 
- This trait contains `get_currency_unit` method. This method needs to be implemented. 
- It will define what type of conversion needs to be done during `connector_router_data` creation. Concurrently handle the creation of `connector_router_data` to be passed for the connector request body creation.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2196

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

**Comment by DostoevskyOnLinux** [comment]
(If I can't get #2232, can I have this one? Thank you!)

**Comment by Hangsaai** [comment]
Hey @Sakilmostak can you assign this to me as I have basis prior knowledge on Rust and could do this easily

**Comment by swangi-kumari** [comment]
Hey @DostoevskyOnLinux I have already assigned #2232 to you. Can you please work on that. For now, we'll leave this issue open to any other interested contributors. You may pick this up once your work on #899 is completed, and if no other contributors have expressed interest on this. I hope that's fine by you.

**Comment by swangi-kumari** [comment]
Hey @DostoevskyOnLinux I have already assigned https://github.com/juspay/hyperswitch/issues/2232 to you. Can you please work on that. For now, we'll leave this issue open to any other interested contributors. You may pick this up once your work on https://github.com/juspay/hyperswitch/issues/2232 is completed, and if no other contributors have expressed interest on this. I hope that's fine by you.

**Comment by saikiranpatil** [comment]
hey @Sakilmostak @swangi-kumari can you assign this to me?

**Comment by SagarDevAchar** [comment]
Hey, I'd like to work on this!
I am waiting for a PR to be approved.

**Comment by shivansh-bhatnagar18** [comment]
Dear Authors, Can you please assign this issue to me?

**Comment by SanchithHegde** [comment]
Sure @SagarDevAchar, I'll assign this to you, seeing as your changes in #2645 are waiting on approval.

@shivansh-bhatnagar18 We assign issues on a first come first serve basis. Plus, since we just assigned you #2244, you can take up another issue once your work on that is complete.

**Comment by swangi-kumari** [comment]
Completed in this [PR](https://github.com/juspay/hyperswitch/pull/2671)
